include/IOUtils.hpp

```diff
@@ -0,0 +1,94 @@
+#ifndef IO_UTILS_HPP
+#define IO_UTILS_HPP
+
+#include <mpi.h>
+
+#include <cerrno>
+#include <cmath>
+#include <iomanip>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+
+// POSIX mkdir
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include <deal.II/base/utilities.h>
+
+namespace io_utils
+{
+    inline void ensure_directory_exists(const std::string &dir,
+                                        const unsigned int mpi_rank)
+    {
+        if (dir.empty() || dir == ".")
+            return;
+
+        int ok = 1;
+
+        if (mpi_rank == 0)
+        {
+            errno = 0;
+            const int rc = ::mkdir(dir.c_str(), 0755);
```

Comment:
Have you considered using the C++-17 standard library header [`filesystem`](https://en.cppreference.com/w/cpp/filesystem.html) instead?